frontend/app/services/MembersDataAPI.scala

```diff
@@ -93,29 +91,29 @@ object MembersDataAPI {
       case _ => Logger.error(s"Unexpected credentials for getAttributes! ${memberRequest.user.credentials}")
     }
 
-    def upsertBehaviour(request: AuthRequest[_], activity: Option[String] = None, note: Option[String] = None, emailed: Option[Boolean] = None) = {
-      request.user.credentials match {
+    def upsertBehaviour(user: AuthenticatedIdUser, activity: Option[String] = None, note: Option[String] = None, emailed: Option[Boolean] = None) = {
+      user.credentials match {
         case cookies: AccessCredentials.Cookies =>
-          setBehaviour(Seq(request.cookies.get("GU_U"), request.cookies.get("SC_GU_U")), request.user.id, activity, note).onComplete {
-            case Success(result) => Logger.info(s"Upserted ${request.user.id}")
-            case Failure(err) => Logger.error(s"Failed to upsert membership-data-api behaviour for user ${request.user.id}", err)
+          setBehaviour(cookies, user.id, activity, note).onComplete {
+            case Success(result) => Logger.info(s"Upserted ${user.id}")
+            case Failure(err) => Logger.error(s"Failed to upsert membership-data-api behaviour for user ${user.id}", err)
           }
-        case _ => Logger.error(s"Unexpected credentials for addBehaviour ($activity) for ${request.user.credentials}")
+        case _ => Logger.error(s"Unexpected credentials for addBehaviour ($activity) for ${user.credentials}")
       }
     }
 
-    def removeBehaviour(memberRequest: SubReqWithSub[_], activity: Option[String] = None) = memberRequest.user.credentials match {
+    def removeBehaviour(user: AuthenticatedIdUser, activity: Option[String] = None, note: Option[String] = None) = user.credentials match {
       case cookies: AccessCredentials.Cookies =>
-        deleteBehaviour(Seq(memberRequest.cookies.get("GU_U"), memberRequest.cookies.get("SC_GU_U")), memberRequest.user.id, activity).onComplete {
-          case Success(result) => Logger.info(s"Cleared behaviours for ${memberRequest.user.user.id}")
-          case Failure(err) => Logger.error(s"Failed to remove behaviour events via membership-data-api for user ${memberRequest.user.id}", err)
+        setBehaviour(cookies, user.id, activity, note).onComplete {
```

Comment:
Fixed up the errors I made in `deleteBehaviour` with a new version of this commit: 8ec8a40c25f39535d3bc3c9c7e74f25661eb9644